extensions/patterns/configure_ec2/template_rhdh/create_ec2_instance.yml

```diff
@@ -0,0 +1,180 @@
+    - title: Survey
+      required:
+        - aws_region
+        - instance_name
+        - instance_type
+      description: Create EC2 instance in the Amazon cloud
+      properties:
+        aws_region:
+          title: AWS Region
+          description: AWS region where resources should be created
+          type: string
+          ui:options:
+            rows: 5
+        instance_name:
+          title: Instance Name
+          description: Name of EC2 instance to create
+          type: string
+          ui:options:
+            rows: 5
+        instance_type:
+          title: Instance Type
+          description: Type of EC2 instance to create (e.g., t2.micro, m5.large)
+          type: string
+          ui:options:
+            rows: 5
+        ami_id:
+          title: AMI ID
+          description: Amazon Machine Image (AMI) ID to use for the instance, if not provided will default to the RHEL 9 AMI for the provided region and instance type
+          type: string
+          ui:options:
+            rows: 5
+        key_name:
+          title: Key Pair Name
+          description: Name of key pair to use or create for SSH access to the EC2 instance. Defaults to '{{ instance_name }}-key'
+          type: string
+          ui:options:
+            rows: 5
+        wait_for_state:
+          title: Wait for State
+          description: Whether to wait for the EC2 instance to be in the running state before continuing. Defaults to true
+          type: string
+          enum:
+            - ""
+            - 'true'
+            - 'false'
+        instance_tags:
+          title: Instance Tags
+          description: 'A dict of tags for the instance, e.g. {"environment: test", "owner": "team foobar"}'
+          type: string
+          ui:options:
+            rows: 5
+        vpc_name:
+          title: VPC Name
+          description: Name of the VPC to create. Defaults to '{{instance_name}}-vpc'
+          type: string
+          ui:options:
+            rows: 5
+        vpc_cidr:
+          title: VPC CIDR Block
+          description: CIDR block to use for the VPC being created. Defaults to 10.0.0.0/24
+          type: string
+          ui:options:
+            rows: 5
+        subnet_cidr:
+          title: Subnet CIDR block
+          description: CIDR block to use for the subnet being created. 10.0.0.0/25
+          type: string
+          ui:options:
+            rows: 5
+        sg_name:
+          title: Security Group Name
+          description: Name of the security group to create for securing traffic to the instance. Defaults to '{{ instance_name }}-sg'
+          type: string
+          ui:options:
+            rows: 5
+        sg_description:
+          title: Security Group Description
+          description: Description for the security group. Defaults to 'Security group for EC2 instance {{ instance_name }}'
+          type: string
+          ui:options:
+            rows: 5
+        sg_rules:
```

Comment:
One thing we should double check, because I didn't see it tested in the demo video, is that this field correctly handles multiline yaml text, e.g.:
```
- proto: tcp 
  ports: 80 
  cidr_ip: 0.0.0.0/0
```
I've verified this works when running the job from AAP, I just want to make sure it also works as expected when launching the job template from RHDH.

Reply:
I tested, with port 80, 8080, 22 in the list. Part of AAP job output:

```
TASK [Get security group rules list from string input] *************************  
ok: [localhost] => {"ansible_facts": {"sg_rules_list": [{"cidr_ip": "0.0.0.0/0", "ports": 80, "proto": "tcp"}, {"cidr_ip": "0.0.0.0/0", "ports": 8080, "proto": "tcp"}, {"cidr_ip": "0.0.0.0/0", "ports": 22, "proto": "tcp"}]}, "changed": false}
```

It works